core/base/mtx_io.cpp

```diff
@@ -760,6 +763,187 @@ matrix_data<ValueType, IndexType> read_raw(std::istream& is)
 }
 
 
+template <typename ValueType, typename IndexType>
+static constexpr uint64_t binary_format_magic()
```

Comment:
I think this function could use a bit of documentation. It has 'magic' in the name, which usually means something not obvious is happening, so I think it would be reasonable to document it.